include/swift/AST/Decl.h

```diff
@@ -3900,8 +3906,16 @@ class ClassDecl final : public NominalTypeDecl {
   /// initializers that cannot be represented in Swift.
   bool hasMissingDesignatedInitializers() const;
 
-  void setHasMissingDesignatedInitializers(bool newValue = true) {
-    Bits.ClassDecl.HasMissingDesignatedInitializers = newValue;
+  void setHasMissingDesignatedInitializers(bool value = true) {
+    LazySemanticInfo.HasMissingDesignatedInitializers = value;
+    LazySemanticInfo.HasMissingDesignatedInitializersComputed = true;
+
+    // Add an attribute for printing
+    if (value && !getAttrs().hasAttribute<HasMissingDesignatedInitsAttr>()) {
```

Comment:
This doesn't handle *removing* the attribute if it's set to `false`. There's no real reason to implement that, but maybe assert that it's not supported?